src/filter/common.rs

```diff
@@ -0,0 +1,6 @@
+use crate::walk::DirEntry;
+
+pub trait Filter {
+    /// Whether the entry should be skipped or not.
+    fn should_skip(&self, entry: &DirEntry) -> bool;
```

Comment:
Maybe this should return a `WalkState` instead of a bool.  In that case the method name should be different, maybe `filter()`.

Reply:
I think it should stay as `bool` because returning `WalkState` makes one think that all three enum variants (`Continue`, `Skip` and `Quit`) could be returned from the function, when actually we're only talking about whether we should `Skip` or `Continue`.